pyp2spec/license_processor.py

```diff
@@ -0,0 +1,146 @@
+                fedora_identifier = trove2fedora_map[classifier]
+            except KeyError:
+                err_string = """{0}: Such classifier doesn't exist.
+                If you believe that's pyp2spec's error, open an issue at project's GitHub repo."""
+                click.secho(err_string.format(classifier), fg="red")
+                sys.exit(1)
+            # not every classifier maps unambiguously to a signle Fedora-SPDX expression
+            if not fedora_identifier:
+                click.secho(
+                    f"'{classifier}' can't be reliably transformed to SPDX", fg="red"
+                )
+                return None
+            # OSI/DFSG Approved are umbrella categories which don't bring
+            # any valuable information, let's just ignore them when encountered
+            # Src: https://peps.python.org/pep-0639/#mapping-license-classifiers-to-spdx-identifiers
+            if fedora_identifier in ("OSI Approved", "DFSG approved"):
+                continue
+            spdx_identifiers.append(fedora_identifier)
+        return spdx_identifiers
+
+    def spdx_expression(self):
+        """Return an approximate SPDX expression (all identifiers joined with operator 'AND').
+        """
+        if not self.spdx_identifiers:
+            raise ValueError
+        return " AND ".join(self.spdx_identifiers)
+
+
+class LicenseMetadataConverter:
+    """Takes care of package `license` keyword value conversion to the Fedora format.
+    """
+
+    def __init__(self, license_expression):
+        self.license_expression = license_expression
+        self.spdx_identifiers = self.split_to_identifiers()
+
+    def is_valid_spdx_expression(self):
+        """Return True if the license expression matches the SPDX rules, otherwise False."""
+        spdx_pattern = r"^\S+$|\(?\S+(\s(AND|OR|WITH)\s\(?\S+\)?)+"
+        if re.fullmatch(spdx_pattern, self.license_expression):
+            return True
+        click.secho(
+            f"'{self.license_expression}' is not a valid SPDX expression", fg="red"
+        )
+        return False
+
+    def split_to_identifiers(self):
+        """Return a list of SPDX identifiers extracted from the license expression.
+        """
+        operators = ("AND", "OR", "WITH")
+        if self.is_valid_spdx_expression():
+            return [
+                re.sub(r"\(|\)", "", i)
+                for i in self.license_expression.split(" ")
+                if i not in operators
+            ]
+
+    def spdx_expression(self):
+        """Return the license expression (if it's a valid SPDX one).
+        """
+        if not self.is_valid_spdx_expression():
+            raise ValueError
+        return self.license_expression
+
+
+def load_fedora_licenses(session):
+    """Load the list of licenses evaluated for Fedora by Fedora Legal team.
+    """
+
+    url = "https://gitlab.com/fedora/legal/fedora-license-data/-/jobs/artifacts/main/raw/fedora-licenses.json?job=json"
```

Comment:
In Fedora, I'd rather load this json from the Fedora package. Is that possible/planned?

Reply:
This feature is added now. First, there's an attempt to read the data from the local file, and only when not found - on the internetz.